src/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -194,6 +194,11 @@ private ApiEvent InitializeApiEvent(string accountId)
 
         protected AuthenticationResult CacheTokenResponseAndCreateAuthenticationResult(MsalTokenResponse msalTokenResponse)
         {
+            if(AuthenticationRequestParameters.IsBrokerEnabled)
```

Comment:
>if(AuthenticationRequestParameters.IsBrokerEnabled) [](start = 12, length = 51)

this can be moved to the function and simply ensure nothing is done if this is not broker. that way the function guards itself